experimental/FTheoryTools/TateModels/attributes.jl

```diff
@@ -0,0 +1,348 @@
+#####################################################
+# 1: The Tate sections
+#####################################################
+
+@doc Markdown.doc"""
+    tate_section_a1(t::GlobalTateModel)
+
+Return the Tate section ``a_1``.
+
+```jldoctest
+julia> t = global_tate_model(test_base())
+Global Tate model over a concrete base
+
+julia> tate_section_a1(t);
+```
+"""
+@attr MPolyElem{fmpq} tate_section_a1(t::GlobalTateModel) = t.a1
+
+
+@doc Markdown.doc"""
+    tate_section_a2(t::GlobalTateModel)
+
+Return the Tate section ``a_2``.
+
+```jldoctest
+julia> t = global_tate_model(test_base())
+Global Tate model over a concrete base
+
+julia> tate_section_a2(t);
+```
+"""
+@attr MPolyElem{fmpq} tate_section_a2(t::GlobalTateModel) = t.a2
+
+
+@doc Markdown.doc"""
+    tate_section_a3(t::GlobalTateModel)
+
+Return the Tate section ``a_3``.
+
+```jldoctest
+julia> t = global_tate_model(test_base())
+Global Tate model over a concrete base
+
+julia> tate_section_a3(t);
+```
+"""
+@attr MPolyElem{fmpq} tate_section_a3(t::GlobalTateModel) = t.a3
+
+
+@doc Markdown.doc"""
+    tate_section_a4(t::GlobalTateModel)
+
+Return the Tate section ``a_4``.
+
+```jldoctest
+julia> t = global_tate_model(test_base())
+Global Tate model over a concrete base
+
+julia> tate_section_a4(t);
+```
+"""
+@attr MPolyElem{fmpq} tate_section_a4(t::GlobalTateModel) = t.a4
+
+
+@doc Markdown.doc"""
+    tate_section_a6(t::GlobalTateModel)
+
+Return the Tate section ``a_6``.
+
+```jldoctest
+julia> t = global_tate_model(test_base())
+Global Tate model over a concrete base
+
+julia> tate_section_a6(t);
+```
+"""
+@attr MPolyElem{fmpq} tate_section_a6(t::GlobalTateModel) = t.a6
```

Comment:
`MPolyElem` is the old name, from before the big renaming. I think you should run the renaming script from https://github.com/oscar-system/Oscar.jl/wiki/The-Big-Renaming on this PR before we merge it. (Beware: that script can make some unwanted changes; I usually use `git add -p` resp. `git checkout -p` to selectively add/discard the changes it makes. A bit tedious but still only takes a few minutes.

Reply:
Second issue: `MPolyElem{fmpq}` is not a concrete type, so it doesn't give type stability to prescribe this type. Instead this should use `QQMPolyRingElem`:
```suggestion
@attr QQMPolyRingElem tate_section_a6(t::GlobalTateModel) = t.a6
```
There are bunch of other places with the same problem.

Reply:
I have applied the renaming script. This seemed to worked quite well. (I will push the changes shortly and we can see if the tests pass. I very much do hope they do.)

I have not yet worked on the second issue that you raise. The renaming script replaced the above line with
```julia
@attr MPolyRingElem{QQFieldElem} tate_section_a6(t::GlobalTateModel) = t.a6
```
I asume you propose to remove `QQFieldElem`, i.e. modify it further to look like this:
```julia
@attr MPolyRingElem tate_section_a6(t::GlobalTateModel) = t.a6
```
Correct?